packages/ui/components/badge/Badge.tsx

```diff
@@ -1,62 +1,58 @@
+import type { VariantProps } from "class-variance-authority";
+import { cva } from "class-variance-authority";
+import type { ComponentProps, ReactNode } from "react";
 import { GoPrimitiveDot } from "react-icons/go";
 
 import classNames from "@calcom/lib/classNames";
 import type { SVGComponent } from "@calcom/types/SVGComponent";
 
-const badgeClassNameByVariant = {
-  default: "bg-orange-100 text-orange-800",
-  warning: "bg-orange-100 text-orange-800",
-  orange: "bg-orange-100 text-orange-800",
-  success: "bg-green-100 text-green-800",
-  green: "bg-green-100 text-green-800",
-  gray: "bg-gray-100 text-gray-800 dark:bg-darkgray-200 dark:text-darkgray-800 group-hover:bg-gray-200 dark:group-hover:bg-darkgray-300",
-  blue: "bg-blue-100 text-blue-800",
-  red: "bg-red-100 text-red-800",
-  error: "bg-red-100 text-red-800",
-};
+const badgeStyles = cva("font-medium inline-flex items-center justify-center rounded gap-x-1", {
+  variants: {
+    variant: {
+      default: "bg-orange-100 text-orange-800",
+      warning: "bg-orange-100 text-orange-800",
+      orange: "bg-orange-100 text-orange-800",
+      success: "bg-green-100 text-green-800",
+      green: "bg-green-100 text-green-800",
+      gray: "bg-gray-100 text-gray-800 dark:bg-darkgray-200 dark:text-darkgray-800 group-hover:bg-gray-200 dark:group-hover:bg-darkgray-300",
+      blue: "bg-blue-100 text-blue-800",
+      red: "bg-red-100 text-red-800",
+      error: "bg-red-100 text-red-800",
+    },
+    size: {
+      sm: "px-1 py-0.5 text-xs",
+      md: "py-1 px-1.5 text-xs",
+      lg: "py-1 px-2 text-sm",
+    },
+  },
+  defaultVariants: {
+    variant: "default",
+    size: "sm",
+  },
+});
 
-const classNameBySize = {
-  default: "h-5",
-  lg: "h-6",
-};
+type InferredBadgeStyles = VariantProps<typeof badgeStyles>;
+
+type IconOrDot =
+  | {
+      startIcon?: SVGComponent;
+      withDot?: unknown;
+    }
+  | { startIcon?: unknown; withDot?: boolean };
 
-export type BadgeProps = {
-  variant: keyof typeof badgeClassNameByVariant;
-  size?: keyof typeof classNameBySize;
-  StartIcon?: SVGComponent;
-  bold?: boolean;
-  withDot?: boolean;
-  rounded?: boolean;
-} & JSX.IntrinsicElements["div"];
+export type BadgeProps = InferredBadgeStyles &
+  ComponentProps<"div"> & { children: ReactNode; rounded?: boolean } & IconOrDot;
 
 export const Badge = function Badge(props: BadgeProps) {
-  const {
-    variant = "default",
-    className,
-    size = "default",
-    rounded,
-    StartIcon,
-    withDot,
-    bold,
```

Comment:
The `bold` prop got removed but there is still a caller using it in `[uid].tsx` (It's currently preventing the build from passing). Was the removal intentional and we can clean up that `bold` or do we need it still?

Reply:
> Changes look solid but we need to decide on the `bold` prop fix.

i think it's good to go by removing the bold prop because in the DS there isn't a variation with font weights. also fixed the build error.

cc: @calcom/ui